### PR TITLE
feat: Deprecate `gatsby-recipes`

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This moves `gatsby-recipes` into the `deprecated-packages` directory in this monorepo. Thus with the next release when this is merged the package will no longer receive any new publishes and the current version will stay. A possible follow-up for this PR could be changing the README of `gatsby-recipes` to include a snippet like:

```shell
# You can still use it like this
npx gatsby-cli@version recipes emotion
```

I'd need to verify first if `npx` in fact includes supporting specifying a specific version.

### For Review

- 144 files of the files changed are just `gatsby-recipes` being moved to the new directory.
- I've changed the `create-gatsby` README as the instructions were not correct when I tried it out myself
- `create-gatsby` and `gatsby-cli` relied on some functionality from recipes, I've reimplemented it where necessary

### Documentation

The glossary entry was updated and one doc that mentioned a post about using recipes.

## Related Issues

Partially addresses https://github.com/gatsbyjs/gatsby/issues/25764 - more needs to be done but with recipes out one less culprit.

[sc-42424]

### CODE REVIEW

File: packages/gatsby-cli/src/create-cli.ts [hunk 1]
```diff
@@ -386,46 +386,15 @@ function buildLocalCommands(cli: yargs.Argv, isLocalSite: boolean): void {
...
-        .positional(`cmd`, {
-          choices: [`docs`, `ls`],
-          describe: "Valid commands include `docs`, `ls`.",
-          type: `string`,
-        })
-        .positional(`plugins`, {
-          describe: `The plugin names`,
-          type: `string`,
-        }),
+      yargs.positional(`cmd`, {
```
Comment (LekoArts) [inline packages/gatsby-cli/src/create-cli.ts:393]: The stub was added in https://github.com/gatsbyjs/gatsby/pull/27725 but never used again. So `plugins` positional was never working, thus I removed this here + removed `plugin-add.ts` handler file

File: packages/gatsby-cli/src/init-starter.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import isValid from "is-valid-path"
 import sysPath from "path"
 import prompts from "prompts"
 import url from "url"
-import { updateSiteMetadata } from "gatsby-core-utils"
+import { updateInternalSiteMetadata } from "gatsby-core-utils"
```
Comment (LekoArts) [inline packages/gatsby-cli/src/init-starter.ts:12]: The naming of the function confused me with the `siteMetadata` from gatsby-config.js so I renamed this

File: packages/gatsby-core-utils/src/site-metadata.ts [hunk 3]
```diff
@@ -28,3 +31,30 @@ export async function updateSiteMetadata(
...
+  src: string,
+  { name, value }: { name: string; value: string }
+): string {
+  const FIND = `  siteMetadata: {\n`
+  const REPLACE = `  siteMetadata: {\n      ${name}: \`${value}\`,\n`
+  const modifiedConfig = src.replace(FIND, REPLACE)
+  return modifiedConfig
+}
+
+export async function addFieldToMinimalSiteMetadata(
```
Comment (LekoArts) [inline packages/gatsby-core-utils/src/site-metadata.ts:49]: Largely copied from gatsby-recipes

File: packages/gatsby-core-utils/src/site-metadata.ts [hunk 4]
```diff
@@ -28,3 +31,30 @@ export async function updateSiteMetadata(
     unlock => unlock?.()
   )
 }
+
+/**
+ * Does a string replace by searching for beginning of "siteMetadata"
+ * Then it adds the name + value as the next key of that object
+ */
+function addField(
```
Comment (LekoArts) [inline packages/gatsby-core-utils/src/site-metadata.ts:39]: This was previously a babel step, now it's a naive string replace

**Review by wardpeet — APPROVED** [review]
I don't want to spend to long reviewing it. If create-gatsby and gatsby still works as expected i'm good to merge this!

💀 Recipes 💀

**Review by wardpeet — DISMISSED** [review]

**Review by wardpeet — APPROVED** [review]

File: packages/gatsby-cli/src/init-starter.ts (hunk 2 above)
Comment (NickScottKortical) [reply packages/gatsby-cli/src/init-starter.ts:12]: seem to be getting this error when running `gatsby build` now: 

`error (0 , _gatsbyCoreUtils.updateSiteMetadata) is not a function`

```
TypeError: (0 , _gatsbyCoreUtils.updateSiteMetadata) is not a function
  
  - build.ts:87 build
    [project]/[gatsby]/src/commands/build.ts:87:9
  
  - create-cli.js:240 
    [project]/[gatsby-cli]/lib/create-cli.js:240:14
  
  - create-cli.js:130 
    [project]/[gatsby-cli]/lib/create-cli.js:130:24
  
  - create-cli.js:41 Object.handler
    [project]/[gatsby-cli]/lib/create-cli.js:41:19
  
  - command.js:196 Object.runCommand
    [project]/[gatsby-cli]/[yargs]/build/lib/command.js:196:48
  
  - yargs.js:1043 Object.parseArgs [as _parseArgs]
    [project]/[gatsby-cli]/[yargs]/build/lib/yargs.js:1043:55
  
  - yargs.js:584 Object.parse
    [project]/[gatsby-cli]/[yargs]/build/lib/yargs.js:584:29
  
  - create-cli.js:541 createCli
    [project]/[gatsby-cli]/lib/create-cli.js:541:136
  
  - index.js:84 Object.<anonymous>
    [project]/[gatsby-cli]/lib/index.js:84:26
  
  - loader:1101 Module._compile
    node:internal/modules/cjs/loader:1101:14
  
  - loader:1153 Object.Module._extensions..js
    node:internal/modules/cjs/loader:1153:10
  
  - loader:981 Module.load
    node:internal/modules/cjs/loader:981:32
  
  - loader:822 Function.Module._load
    node:internal/modules/cjs/loader:822:12
  
  - loader:1005 Module.require
    node:internal/modules/cjs/loader:1005:19
  
  - helpers:102 require
    node:internal/modules/cjs/helpers:102:18
  
  - gatsby.js:3 Object.<anonymous>
    [project]/[gatsby]/src/bin/gatsby.js:3:1
  

error Command failed with exit code 1.
```

I need to have a look into why this is happening though, as something in our build system isn't correctly pinning the version of gatsby + gatsby-cli it seems. (setting to the previous version doesn't fix the issue for some reason - should have an update tomorrow). I'm guessing (and unsure if this is always the case, but this change means that the gatsby + gatsby-cli versions must match)

Comment (LekoArts) [reply packages/gatsby-cli/src/init-starter.ts:12]: You shouldn’t add gatsby-cli as a dependency to a project. It’s a global CLI package

Comment (NickScottKortical) [reply packages/gatsby-cli/src/init-starter.ts:12]: To be clear we don't have it as a dependency in the package.json, however it's required in the CI system to be installed

**Pull request merged**